File: ZenCare.Services/Database/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ZenCare.Model.Enums;

namespace ZenCare.Services.Database;

public class Product
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(1000)]
    public string? Description { get; set; }

    [Column(TypeName = "decimal(18,2)")]
    public decimal Price { get; set; }

    public int StockQuantity { get; set; }

    [MaxLength(50)]
    public string? SKU { get; set; }

    public ProductStatus Status { get; set; } = ProductStatus.Draft;

    public int ProductCategoryId { get; set; }

    [ForeignKey(nameof(ProductCategoryId))]
    public ProductCategory ProductCategory { get; set; } = null!;

    public int ProductTypeId { get; set; }

    [ForeignKey(nameof(ProductTypeId))]
    public ProductType ProductType { get; set; } = null!;

    public int UnitOfMeasureId { get; set; }

    [ForeignKey(nameof(UnitOfMeasureId))]
    public UnitOfMeasure UnitOfMeasure { get; set; } = null!;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();

    public ICollection<PurchaseItem> PurchaseItems { get; set; } = new List<PurchaseItem>();

    public ICollection<Review> Reviews { get; set; } = new List<Review>();

    public ICollection<RecommendationLog> RecommendationLogs { get; set; } = new List<RecommendationLog>();
}

File: ZenCare.Mobile/lib/models/product.dart
class Product {
  Product({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    required this.stockQuantity,
    required this.productCategoryId,
    required this.productCategoryName,
    required this.productTypeId,
    required this.productTypeName,
    required this.unitOfMeasureId,
    required this.unitOfMeasureName,
    required this.supplierId,
    required this.supplierName,
    required this.isActive,
  });

  final int id;
  final String name;
  final String? description;
  final double price;
  final int stockQuantity;
  final int productCategoryId;
  final String productCategoryName;
  final int productTypeId;
  final String productTypeName;
  final int unitOfMeasureId;
  final String unitOfMeasureName;
  final int supplierId;
  final String supplierName;
  final bool isActive;

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
      price: (json['price'] as num?)?.toDouble() ?? 0,
      stockQuantity: json['stockQuantity'] as int? ?? 0,
      productCategoryId: json['productCategoryId'] as int? ?? 0,
      productCategoryName: json['productCategoryName'] as String? ?? '',
      productTypeId: json['productTypeId'] as int? ?? 0,
      productTypeName: json['productTypeName'] as String? ?? '',
      unitOfMeasureId: json['unitOfMeasureId'] as int? ?? 0,
      unitOfMeasureName: json['unitOfMeasureName'] as String? ?? '',
      supplierId: json['supplierId'] as int? ?? 0,
      supplierName: json['supplierName'] as String? ?? '',
      isActive: json['isActive'] as bool? ?? false,
    );
  }
}
